### PR TITLE
Update orderedselect_input.pt

### PR DESCRIPTION
**Pull request by szakitibi**
Add "required" attribute to <select name="to"> for required fields #168

### CODE REVIEW

**Comment by szakitibi** [comment]
Regarding "Plone Contributors Agreement verifier":
>  Plone Release Manager, the patch is so tiny as not to constitute a "creative work." 
I'm hereby transfer the intellectual property rights to the Plone Foundation for this PR.

**Comment by szakitibi** [comment]
I'm not sure about the PR at all for two reasons:

**Reason 1:** Adding `required` to the `OrderedSelectWidget` is not sufficient fix, since it will still fail validation on backend. Thus it can be hacked by simply removing the `required` attribute. A cosmetic fix could do more harm this way allowing broken data into DB.

**Reason 2.** I've looked into the underlying logic in [`zope.scheam._field.List`](https://github.com/zopefoundation/zope.schema/blob/7.0.1/src/zope/schema/_field.py#L816). Seems like:

- [`Collection`](https://github.com/zopefoundation/zope.schema/blob/7.0.1/src/zope/schema/_field.py#L726) itself validates only that the provided value is from [`value_type` sequence](https://github.com/zopefoundation/zope.schema/blob/7.0.1/src/zope/schema/_field.py#L767-L779).
- [`Iterable`](https://github.com/zopefoundation/zope.schema/blob/7.0.1/src/zope/schema/_bootstrapfields.py#L446) only validates that value is indeed an iterable and not raises error on `iter(value)`. It also calls ['Container`](https://github.com/zopefoundation/zope.schema/blob/7.0.1/src/zope/schema/_bootstrapfields.py#L431) super classes validation checking value has `__contains__`. Then super [`Field`](https://github.com/zopefoundation/zope.schema/blob/7.0.1/src/zope/schema/_bootstrapfields.py#L342) checks type and `None` for required, which the empty list slips through.
-  [`MinMaxLen`](https://github.com/zopefoundation/zope.schema/blob/7.0.1/src/zope/schema/_bootstrapfields.py#L496) validates the length though

Seems like some widgets work by my logic (e.g. `plone.app.z3cform.widget.SelectFieldWidget`) and raise error for empty list. While others use a different logic (e.g. `z3c.form.browser.orderedselect.OrderedSelectWidget`) not raising error and accepting empty list.

No it could happen that I'm assuming something here and the `List` field (even if it is `required=True`) is normal to accept empty list. To get a validation error it should be always pared with `min_length=1`.

Note: If this is the case, then 'SelectWdiget' and 'AjaxSelectWdigets' raises error for no reason?

**Comment by gforcada** [comment]
Hi! First of all thanks for the contribution! 🎉 

I'm unfortunately not an expert on the validation part, but even if some further work might be needed on the validation logic, marking a field as required is a good starting point IMHO.

Maybe some others have a better insight?

/cc @petschki @thet @jensens ? 😄

**Comment by jensens** [comment]
From the logic POV, passing the views required is the right thing to to AFAICT.

File: plone/app/z3cform/templates/orderedselect_input.pt [hunk 1]
```diff
@@ -104,6 +105,7 @@
                   onchange view/onchange;
                   multiple view/multiple;
                   size view/size;
+                  required required | nothing;
```
Comment (petschki) [inline plone/app/z3cform/templates/orderedselect_input.pt:108]: I'd follow best practice of the other widget templates and use a onelineer

```suggestion
                  required python:view.required and 'required' or None;
```

and remove the `tal:define` line above.

**Review by petschki — CHANGES_REQUESTED** [review]

**Comment by petschki** [comment]
@szakitibi thanks for this one ... please see my successor with your changes. As I've mentioned you in the changenote, you might feel motivated to sign our contributors agreement. It's definitely worth it 😉

**Pull request closed without merging**